packages/contracts/test/transactionManager.spec.ts

```diff
@@ -616,17 +736,167 @@ describe("TransactionManager", function () {
+
+      const variant = {
+        amount: record.amount,
+        expiry: record.expiry,
+        preparedBlockNumber: blockNumber,
+      };
+
+      const signature = await signFulfillTransactionPayload(transaction, relayerFee, router);
+
+      await expect(
+        transactionManager.connect(router).fulfill(
+          {
+            ...transaction,
+            ...variant,
+          },
+          relayerFee,
+          signature,
+        ),
+      ).to.be.revertedWith("fulfill: INVALID_SIGNATURE");
+    });
+
+    it("should revert if relayer fee is higher than amount", async () => {
+      const { transaction, record } = await getTransactionData();
+      const relayerFee = "10";
+      const { blockNumber } = await prepareAndAssert(transaction, record, user, transactionManager);
+
+      const variant = {
+        amount: record.amount,
+        expiry: record.expiry,
+        preparedBlockNumber: blockNumber,
+      };
+
+      const signature = await signFulfillTransactionPayload(transaction, relayerFee, user);
+
+      await expect(
+        transactionManager.connect(router).fulfill(
+          {
+            ...transaction,
+            ...variant,
+          },
+          relayerFee,
+          signature,
+        ),
+      ).to.be.revertedWith("fulfill: INVALID_RELAYER_FEE");
+    });
+
+    it("should revert iff it's sending chain and interacting party is not router", async () => {
+      const { transaction, record } = await getTransactionData();
+      const relayerFee = "1";
+      const { blockNumber } = await prepareAndAssert(transaction, record, user, transactionManager);
+
+      const variant = {
+        amount: record.amount,
+        expiry: record.expiry,
+        preparedBlockNumber: blockNumber,
+      };
+
+      const signature = await signFulfillTransactionPayload(transaction, relayerFee, user);
+
+      await expect(
+        transactionManager.connect(user).fulfill(
+          {
+            ...transaction,
+            ...variant,
+          },
+          relayerFee,
+          signature,
+        ),
+      ).to.be.revertedWith("fulfill: ROUTER_MISMATCH");
+    });
 
-    it.skip("should revert iff it's receiving chain and relayer fee is non zero and transfer failed", async () => {});
-    it.skip("should revert iff it's receiving chain and calldata is non zero bytes and transfer failed", async () => {});
-    it.skip("should revert iff it's receiving chain and calldata is zero bytes and MultisendInterpreter failed", async () => {});
-    it.skip("should revert iff it's receiving chain and calldata is zero bytes and Interpreter isn't approved for respective amount when asset ERC20", async () => {});
+    it.skip("Happy case: iff it's receiving chain and calldata is non-zero bytes and MultisendInterpreter failed", async () => {});
+    it.skip("Happy case: iff it's receiving chain and calldata is non-zero bytes and MultisendInterpreter success", async () => {});
```

Comment:
Important, and we have to check that IF multisender fails, then the fallback address is transferred to (ie. `try-catch` is handled)